Feat/add uart support

Dzarda7 opened the pull request:
## Description

This PR adds support for UART reception and processing of SLIP command. It was tested and works on the following targets:

esp32 v3.1
esp32c2 v2.0 (v1.0 does not work!)
esp32c3 v1.0
esp32c6 v0.1
esp32h2 v0.1
esp32s3 v0.2
esp32s2 v1.0
esp32c5 v1.0 and v1.1
esp32p4 v1.2 (need new target for >=v3.0)

Along the way, other stuff was fixed and resolved to enable this like not linking standard libraries which caused build with warnings for newer targets. Erase of bss section and account for checksum in SLIP header.

## Related

Related esp-stub-lib PR, not merged yet, needs some modifications: https://github.com/espressif/esp-stub-lib/pull/30

## Testing

Tested on mentioned targets using `esptool flash-id` command. Works, some targets fail because flash chip is not attached, but stub responds correctly. 

## Checklist

Before submitting a Pull Request, please ensure the following:

- [X] 🚨 This PR does not introduce breaking changes.
- [X] All CI checks (GH Actions) pass.
- [X] Documentation is updated as needed.
- [ ] Tests are updated or added as necessary.
- [X] Code is well-commented, especially in complex areas.
- [X] Git history is clean — commits are squashed to the minimum necessary.

jedi7 commented on src/main.c:
```diff
@@ -19,22 +23,57 @@ __asm__(
+        }
+
+        if (slip_is_frame_error()) {
+            slip_recv_reset();
```
can we send an error to user?

Dzarda7 replied:
We cannot due to unforeseen effects in esptool for now. Later definitely.

Dzarda7 commented on src/slip.c:
```diff
@@ -95,10 +132,46 @@ void slip_recv_byte(uint8_t byte)
+        // Add escaped byte to buffer
+        if (s_recv_ctx.frame_length < s_recv_ctx.buffer_size) {
+            s_recv_ctx.buffer[s_recv_ctx.frame_length] = byte;
+            s_recv_ctx.frame_length++;
```
Fixed, thanks.

Dzarda7 commented:
@dobairoland @radimkarnis @jedi7 I believe this should be ready, incorporates all the new changes from esp-stub-lib. I addressed all the suggestions. If you agree, I will merge this.
